src/components/forms/UpdatePassword.jsx

```diff
@@ -0,0 +1,211 @@
+import React, { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import 'tailwindcss/tailwind.css';
+import { setCurrentPassword,setConfirmPassword,setError,setLoading,setNewPassword } from "../../redux/reducers/userSlice";
+import {AiOutlineEyeInvisible,AiOutlineEye} from 'react-icons/ai'
+import { updatePassword } from "../../redux/reducers/userSlice";
+
+
+export const UpdatePassword=()=>{
+      const [currentPasswordError,setCurrentPasswordError]=useState('')
+      const [newPasswordError,setNewPasswordError]=useState('')
+      const [confirmPasswordError,setConfirmPasswordError]=useState('')
+      let   [currentOpenEye,setOpenEye]=useState(false)
+      let   [ newOpenEye,setNewOpenEye]=useState(false)
+      let   [confirmOpenEye,setConfirmOpenEye]=useState(false)
+      const dispatch=useDispatch();
+      const navigate=useNavigate();
+      const currentToggle=()=>{
+        setOpenEye(!currentOpenEye)
+      }
+      const newToggle=()=>{
+        setNewOpenEye(!newOpenEye)
+      }
+      const confirmToggle=()=>{
+        setConfirmOpenEye(!confirmOpenEye)
+      }
+
+      const passwordValidations=()=>{
+       let  isPwdValid=true;
+       if(newPassword !==confirmPassword){
+        setConfirmPasswordError("password don't march!")
+        isPwdValid=false;
+       }
+     if(confirmPassword===""){
+        setConfirmPasswordError("fill the field!")
+        isPwdValid=false
+       }
+       else if(newPassword !==confirmPassword){
+        setConfirmPasswordError("password don't march!")
+        isPwdValid=false;
+       }
+       else{
+        setConfirmPasswordError('')
+        isPwdValid=true;
+       }
+       if(currentPassword===""){
+        setCurrentPasswordError("fill the field!")
+        isPwdValid=false
+       }
+       else if(!currentPassword){
+        setCurrentPasswordError('enter valid password')
+        isPwdValid=false
+       }
+       else{
+        setCurrentPasswordError('')
+       }
+       if(newPassword===""){
+        setNewPasswordError("fill the field!")
+        isPwdValid=false
+       }
+       else if (newPassword.length < 8) {
+        setNewPasswordError('Password must be at least 8 characters long');
+        isPwdValid = false;
+      } else {
```

Comment:
Your code is not following ours, like you're not using YUP as we confirmed for validation. you're using state for every kind on action which is making your code unreadable and inconsistency.

Reply:
thanks for the feedbacks